lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -427,6 +428,22 @@ public function fetchAllAssociative(): array
         return FetchUtils::fetchAllAssociative($this);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchAllKeyValue(): array
```

Comment:
Drivers do not implement this method in `3.x`. Why does it need to be implemented in `2.x`?

Reply:
It is not available in the driver implementation, but feature is ensured by the Result class : https://github.com/doctrine/dbal/blob/3.0.x/src/Result.php#L120-L131

It seems to me we have to ensure that user can write the following :
```php
$result = $connection->executeQuery(...)->fetchAllKeyValue();
```

Reply:
It creates a BC break in 3.0.0 because these methods won't exist there. The driver layer must implement only a subset of `fetch*()` methods and none of the `iterate*()` ones.